lib/gs1/record.rb

```diff
@@ -48,8 +47,8 @@ def to_ai
     end
 
     def ==(other)
-      self.class == other.class &&
-        to_s == other.to_s
+      self.class.equal?(other.class) &&
```

Comment:
Please explain this change

Reply:
from commit message: 

equal?() tests object identity as well.
== sometimes includes type conversion where eql?() does not.

it was part of the feedback from mutation testing